test/sciml_style.jl

```diff
@@ -430,61 +430,153 @@
-                arg4, arg5)
-            body
-        end
-        """
-        @test format_text(str, SciMLStyle(), margin = 36) == fstr
-        # should be 30? might be a unnesting off by 1 error
-        @test format_text(str, SciMLStyle(), margin = 31) == fstr
-
-        fstr = """
-        function foo(
+        @testset "vector definition" begin
+            str = """
+            test = [arg1, arg2, arg3, arg4, arg5]
+            """
+
+            # Fits within the margin
+            @test format_text(str, SciMLStyle(), margin = 41) == str
+            @test format_text(str, SciMLStyle(), margin = 37) == str
+
+            fstr = """
+            test = [
+                arg1, arg2, arg3, arg4, arg5]
+            """
+            @test format_text(str, SciMLStyle(), margin = 36) == fstr
+            @test format_text(str, SciMLStyle(), margin = 33) == fstr
+
+            fstr = """
+            test = [arg1, arg2, arg3,
+                arg4, arg5]
+            """
+            @test format_text(str, SciMLStyle(), margin = 32) == fstr
+            # should be 25? might be a unnesting off by 1 error
+            @test format_text(str, SciMLStyle(), margin = 26) == fstr
+
+            fstr = """
+            test = [
                 arg1, arg2, arg3,
-                arg4, arg5)
-            body
-        end
-        """
-        @test format_text(str, SciMLStyle(), margin = 29) == fstr
-        @test format_text(str, SciMLStyle(), margin = 25) == fstr
+                arg4, arg5]
+            """
+            @test format_text(str, SciMLStyle(), margin = 25) == fstr
+            @test format_text(str, SciMLStyle(), margin = 21) == fstr
 
-        fstr = """
-        function foo(
+            fstr = """
+            test = [arg1, arg2,
+                arg3,
+                arg4, arg5]
+            """
+            @test format_text(str, SciMLStyle(), margin = 20) == fstr
+            # should be 19? might be a unnesting off by 1 error
+            # @test format_text(str, SciMLStyle(), margin = 19) == fstr
+
+            fstr = """
+            test = [
                 arg1, arg2,
                 arg3,
-                arg4, arg5)
-            body
-        end
-        """
-        @test format_text(str, SciMLStyle(), margin = 24) == fstr
+                arg4, arg5]
+            """
+            @test format_text(str, SciMLStyle(), margin = 19) == fstr
+            # should be 15? might be a unnesting off by 1 error
+            @test format_text(str, SciMLStyle(), margin = 16) == fstr
+
+            fstr = """
+            test = [
+                arg1, arg2,
+                arg3,
+                arg4,
+                arg5]
+            """
+            @test format_text(str, SciMLStyle(), margin = 15) == fstr
 
-        fstr = """
-        function foo(
+            fstr = """
+            test = [arg1,
+                arg2,
+                arg3,
+                arg4,
+                arg5]
+            """
```

Comment:
These look a bit weird. But this is how it's formatted on master. I only added these tests here to check that best optimal nesting still works with vector definitions because my previous attempt broke this.